tests/python/relay/test_type_infer.py

```diff
@@ -460,6 +460,36 @@ def test_add_multiple_with_type_var():
     assert mod[list_id].checked_type == relay.FuncType([l(a)], l(a), [a])
 
 
+def test_add_multiple_with_type_var_nonrecursive():
+    mod = relay.Module()
+    p = Prelude(mod)
+    add_nat_definitions(p)
+    l, nat, nil, cons, hd = p.l, p.nat, p.nil, p.cons, p.hd
+    optional, some, none = p.optional, p.some, p.none
+
+    hd_hd = relay.GlobalVar('hd_hd')
+    a = relay.TypeVar('a')
+    x = relay.Var('x', l(l(a))) # fails without this annotation
```

Comment:
In particular, if I leave off the annotation here, I am left with an unresolved incomplete type in the type for main that does not get caught by the resolver! This suggests some serious holes in the later stages of type checking @MarisaKirisame @jroesch